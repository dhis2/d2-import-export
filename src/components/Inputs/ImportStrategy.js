import React from 'react'
import i18n from '@dhis2/d2-i18n'
import { RadioGroupField } from '../index'

const importStrategyOptions = [
    {
        value: 'CREATE_AND_UPDATE',
        label: i18n.t('Import new values and update existing'),
    },
    {
        value: 'CREATE',
        label: i18n.t('Import new values only'),
    },
    {
        value: 'UPDATES',
        label: i18n.t('Only update existing values, ignore new values'),
    },
    {
        value: 'DELETE',
        label: i18n.t('Remove values not included in uploaded file'),
    },
]
const defaultImportStrategyOption = importStrategyOptions[0].value

const NAME = 'importStrategy'
const DATATEST = 'input-import-strategy'
const LABEL = i18n.t('Import strategy')

const ImportStrategy = () => (
    <RadioGroupField
        name={NAME}
        label={LABEL}
        options={importStrategyOptions}
        dataTest={DATATEST}
        vertical
    />
)

export { ImportStrategy, defaultImportStrategyOption }
