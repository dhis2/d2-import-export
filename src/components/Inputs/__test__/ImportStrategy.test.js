import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { ImportStrategy } from '../ImportStrategy'

jest.mock('react-final-form')

describe('Input component - ImportStrategy', () => {
    it('should render correctly', () => {
        const file = shallow(<ImportStrategy />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
