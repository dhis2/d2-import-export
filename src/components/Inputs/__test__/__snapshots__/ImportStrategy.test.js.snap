// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - ImportStrategy should render correctly 1`] = `
<Field>
  <Label>
    Import strategy
  </Label>
  <RadioGroup
    dataTest="input-import-stragegy"
    label="Import strategy"
    name="importStrategy"
    options={
      Array [
        Object {
          "label": "New and updates",
          "value": "CREATE_AND_UPDATE",
        },
        Object {
          "label": "New only",
          "value": "CREATE",
        },
        Object {
          "label": "Updates only",
          "value": "UPDATES",
        },
        Object {
          "label": "Delete",
          "value": "DELETE",
        },
      ]
    }
  />
</Field>
`;
